src/Compilers/CSharp/Portable/Compilation/CSharpDiagnosticFilter.cs

```diff
@@ -214,7 +217,17 @@ internal static class CSharpDiagnosticFilter
 
             if (!isSpecified)
             {
-                report = isEnabledByDefault ? ReportDiagnostic.Default : ReportDiagnostic.Suppress;
+                if (errorCode == (int)ErrorCode.WRN_Experimental)
+                {
+                    // Special handling for [Experimental] warning (treat as error severity by default)
+                    severity = DiagnosticSeverity.Error;
+                    report = ReportDiagnostic.Error;
+                    isSpecified = true;
```

Comment:
Why do we need to set `isSpecified = true` here?

Reply:
You're right, it's not necessary/useful